domain/schema_validator.go

```diff
@@ -80,52 +77,45 @@ func (s *schemaValidator) Restart() {
 	log.Info("the schema validator restarts")
 	s.mux.Lock()
 	defer s.mux.Unlock()
-	s.validItems = make(map[int64]time.Time)
-	s.cacheItemInfos = make(map[int64]*deltaSchemaInfo)
+	s.detalItemInfos = make(map[int64]*deltaSchemaInfo)
 }
 
 func (s *schemaValidator) Update(leaseGrantTS uint64, oldVer, currVer int64, changedTableIDs []int64) {
 	s.mux.Lock()
+	defer s.mux.Unlock()
 
-	if s.cacheItemInfos == nil {
-		s.mux.Unlock()
+	if s.detalItemInfos == nil {
 		log.Infof("the schema validator stopped before updating")
 		return
 	}
 
 	// Renew the lease.
 	leaseGrantTime := extractPhysicalTime(leaseGrantTS)
 	leaseExpire := leaseGrantTime.Add(s.lease - time.Millisecond)
-	s.validItems[currVer] = leaseExpire
-	_, hasCurrVerItemInfo := s.cacheItemInfos[currVer]
+	_, hasCurrVerItemInfo := s.detalItemInfos[currVer]
 	// Update the schema information.
-	if currVer != oldVer || currVer == 0 || !hasCurrVerItemInfo {
-		s.cacheItemInfos[currVer] = &deltaSchemaInfo{
+	if currVer != oldVer || !hasCurrVerItemInfo {
+		s.detalItemInfos[currVer] = &deltaSchemaInfo{
 			expire:          leaseExpire,
 			relatedTableIDs: changedTableIDs,
 		}
-		oldInfo, ok := s.cacheItemInfos[oldVer]
+		oldInfo, ok := s.detalItemInfos[oldVer]
 		if ok {
 			oldInfo.nextSchemaVersion = currVer
 		}
 	} else {
-		s.cacheItemInfos[currVer].expire = leaseExpire
+		s.detalItemInfos[currVer].expire = leaseExpire
 	}
 	s.latestSchemaVer = currVer
-	s.latestSchemaInfo = s.cacheItemInfos[currVer]
-
-	// Delete expired cacheItemInfos, leaseGrantTime is server current time, actually.
-	for k, info := range s.cacheItemInfos {
-		if leaseGrantTime.After(info.expire) {
-			delete(s.validItems, k)
-			// We cache some expired schema versions to store recently updated table IDs.
-			if shouldUpdateAllSchema(currVer, k) {
-				delete(s.cacheItemInfos, k)
-			}
+	s.latestSchemaInfo = s.detalItemInfos[currVer]
+
+	// Delete expired detalItemInfos, leaseGrantTime is server current time, actually.
+	for k, info := range s.detalItemInfos {
+		// We cache some expired schema versions to store recently updated table IDs.
+		if leaseGrantTime.After(info.expire) && shouldUpdateAllSchema(currVer, k) {
```

Comment:
This means we seldom clean items in detaItemInfos, would that map take huge memory?